src/pages/KeywordPage.jsx

```diff
@@ -20,21 +20,20 @@ const KeywordPage = () => {
 
   const { groupId, keywordId } = useParams();
   const [dashboardType, setDashboardType] = useState("chart");
-  const [filterList, setFilterList] = useState({
-    includedKeyword: [],
-    excludedKeyword: [],
-  });
+  const [filterList, setFilterList] = useState(POST_LISTS.DEFAULT_FILTER_LIST);
   const setUserGroupList = useBoundStore((state) => state.setUserGroupList);
   const userUid = useBoundStore((state) => state.userInfo.uid);
   const hasUserUid = !!userUid;
   const hasKeywordId = !!keywordId;
+  const resetFilterList = () => {
+    setFilterList(POST_LISTS.DEFAULT_FILTER_LIST);
+  };
 
   useEffect(() => {
-    setFilterList(() => ({
-      includedKeyword: [],
-      excludedKeyword: [],
-    }));
-  }, [keywordId, setFilterList]);
+    if (keywordId) {
+      resetFilterList();
+    }
+  }, [keywordId]);
```

Comment:
```suggestion
    if (hasKeywordId) {
      resetFilterList();
    }
  }, [keywordId]);
```
if문의 조건문이 직관적이지 않아 이미 선언되어있는 `hasKeywordId`를 사용하거나 또는 keywordId가 변경될 때마다 초기화해야 하는 것이면 if문을 사용하지 않은데, 어떻게 생각하시나요?

Reply:
```suggestion
  useEffect(() => {
    if (keywordId !== null && keywordId !== undefined) {
      resetFilterList();
    }
  }, [keywordId]);
```
말씀 주신 것 처럼, 조건을 보다 명확하게 수정해보면 좋겠습니다. 한 가지 고려할 사항은, 의존성 배열 관련 내용입니다.

코드의 의도는 `keywordId`가 변경될 때(키워드 간 대시보드 이동) 마다 필터를 초기화 하기 위함입니다. 따라서 `hasKeywordId`로 교체 후 의존성 배열에 `hasKeywordId`를 명시하게 되면 키워드 존재 여부는 변경이 없기 때문에 본래 의도대로 동작하지 않더라구요.

그래서 위 제안 코드처럼 `keywordId` 조건을 구체적으로 명시해도 의도대로 동작합니다. 이대로 수정해보면 어떨까요?

Reply:
네네 좋아요!

Reply:
구두로 논의한 것 처럼, 아래 방식으로 수정했습니다!
```js
useEffect(() => {
    if (keywordId !== null && keywordId !== undefined) {
      resetFilterList();
    }
  }, [keywordId]);
```

[참고 - 의존성 배열에 들어가는 값은 선택할 수 없다](https://ko.react.dev/reference/react/useEffect#specifying-reactive-dependencies)

Reply:
저는 useEffect 내에 사용된 state, props만 의존성 배열에 포함되어야하며 일단 변수는 포함되지 않는다라고 알고 있었는데 덕분에 새롭게 알게되었습니다!